src/frontend/org/voltdb/utils/PBDSegment.java

```
@@ -305,36 +312,49 @@ ExportSequenceNumberTracker scan(BinaryDeque.BinaryDequeScanner scanner) throws
      * itself generally lacks context to decide whether it's final or not.
      *
      * @param isFinal   true if segment is set to final, false otherwise
-     * @throws IOException
      */
 
-    public void setFinal(boolean isFinal) throws IOException {
+    public void setFinal(boolean isFinal) {
 
         setFinal(m_file, isFinal);
     }
 
-    public static void setFinal(File file, boolean isFinal) throws IOException {
+    public static void setFinal(File file, boolean isFinal) {
 
-        UserDefinedFileAttributeView view = getFileAttributeView(file);
-        view.write(IS_FINAL_ATTRIBUTE, Charset.defaultCharset().encode(new Boolean(isFinal).toString()));
+        try {
+            UserDefinedFileAttributeView view = getFileAttributeView(file);
+            view.write(IS_FINAL_ATTRIBUTE, Charset.defaultCharset().encode(new Boolean(isFinal).toString()));
+        } catch (Exception e) {
```

I would prefer if not all exceptions were swallowed but instead maybe just IOExceptions and it still might be nice to log those at debug in case somebody ever wonders why this isn't working.

---

Right - restricting handling to IOException